src/renderer/features/fellowship-votes/components/VotingHistoryList.tsx

```diff
@@ -0,0 +1,69 @@
+import { useUnit } from 'effector-react';
+import { memo, useMemo, useState } from 'react';
+
+import { type Chain } from '@/shared/core';
+import { useI18n } from '@/shared/i18n';
+import { useDeferredList } from '@/shared/lib/hooks';
+import { performSearch, toAddress } from '@/shared/lib/utils';
+import { FootnoteText, SearchInput } from '@/shared/ui';
+import { type Vote as VoteType } from '@/domains/collectives';
+import { identityModel } from '../model/identity';
+
+import { Vote } from './Vote';
+import { VotingHistoryListEmptyState } from './VotingHistoryListEmptyState';
+
+type Props = {
+  items: VoteType[];
+  chain: Chain | null;
+  loading?: boolean;
+};
+
+export const VotingHistoryList = memo(({ items, chain, loading }: Props) => {
```

Comment:
Why so many memos here and there? Every new `items` will start render because references for the old and new one won't be the same. Same situation for chain.
`memo` is fine for dumb components that accepts primitive data like string, number, bool and it fails whenever you pass arrays and objects.
Same point for the following `useMemo` we will end up with dozens of memoized data for each new `query`, `items` and `identity`
`useMemo` is fine for heavy operations where we're sure that we will match previously memoized data 
Speaking about heavy operations, why do they even happen inside UI component? Effector model is the best place for that 😉